### PR TITLE
Feat(eos_cli_config_gen): Add schema for management-ssh

### PR DESCRIPTION
**Pull request by KonikaChaurasiya-GSLab**
### Contributor Checklist

- [x] Create schema fragment matching data model described in README.md and README_v4.0.md
  - README.md is most complete with all keys. README_v4.0 includes partial data models after conversion to lists.
  - Schema fragment path is `roles/eos_cli_config_gen/schemas/schema_fragments/<data_model_key>.schema.yml`.
  - Copy line 1-5 from another schema (comments and outer type:dict).
  - Refer to [schema documentation](https://avd.sh/en/devel/docs/input-variable-validation-BETA.html) for syntax and/or use YAML Lint plugin from Redhat in VSCode.
  - Use `convert_types` on value that could be mixed type or misinterpreted like integers and numeric strings.
- [x] If the data model has been converted from wildcard dicts:
  - Add `convert_types: ['dict']` to the schema.
  - Remove `convert_dicts` from the `templates/eos/<>.j2` and `templates/documentation/<>.j2` templates.
- [x] Run `molecule converge -s build_schemas_and_docs` to update schema and documentation.
- [x] Test by running `molecule converge -s eos_cli_config_gen` and verify no errors or changes to generated configs/docs.

### Reviewer Checklist

- Reviewer 1: Carl
  - [X] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
  - [X] Verify that `convert_dicts` has been removed from templates as applicable.
  - [X] Verify no changes to configs/docs on any molecule scenario
  - [X] Verify that CI pass

- Reviewer 2: jonxstill
  - [x] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
  - [x] Verify that `convert_dicts` has been removed from templates as applicable.
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+              display_name: VRF Name
+      idle_timeout:
+        type: int
+        convert_types:
+        - str
+        min: 0
+        max: 86400
+        description: |
+          idle timeout in minutes
+
```
Comment (jonxstill) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml]:
```suggestion
        description: Idle timeout in minutes
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          type: str
+      hostkey:
+        type: dict
+        keys:
+          server:
+            type: list
+            items:
+              type: str
+      enable:
+        type: bool
```
Comment (jonxstill) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml:65]:
```suggestion
        description: Enable SSH daemon
        type: bool
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        type: dict
+        keys:
+          limit:
+            type: int
+            convert_types:
+            - str
+            min: 1
+            max: 100
+            description: |
+              1-100 SSH Connections
```
Comment (jonxstill) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml]:
```suggestion
            description: Maximum total number of SSH sessions to device
```

Comment (jonxstill) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml]: Making the description more descriptive, also the min/max are already defined and visible in the variable table.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+          per_host:
+            type: int
+            convert_types:
+            - str
+            min: 1
+            max: 20
+            description: |
+              1-20 max sessions from a host
```
Comment (jonxstill) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml]:
```suggestion
            description: Maximum number of SSH sessions to device from a single host
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        - dict
+        items:
+          type: dict
+          keys:
+            name:
+              type: str
+              required: true
+              display_name: VRF Name
+            enable:
+              type: bool
```
Comment (jonxstill) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml:97]:
```suggestion
              description: Enable SSH in VRF
              type: bool
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+      log_level:
+        type: str
+        description: |
+          SSH daemon log level
```
Comment (jonxstill) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml]:
```suggestion
        description: SSH daemon log level
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        type: int
+        convert_types:
+        - str
+        min: 0
+        max: 86400
+        description: |
+          idle timeout in minutes
+
+      cipher:
+        type: list
```
Comment (jonxstill) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml]:
```suggestion
      cipher:
        description: Cryptographic ciphers for SSH to use
        type: list
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+        items:
+          type: str
+      key_exchange:
+        type: list
```
Comment (jonxstill) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml]:
```suggestion
      key_exchange:
        description: Cryptographic key exchange methods for SSH to use
        type: list
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+        items:
+          type: str
+      mac:
+        type: list
```
Comment (jonxstill) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml]:
```suggestion
      mac:
        description: Cryptographic MAC algorithms for SSH to use
        type: list
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+        items:
+          type: str
+      hostkey:
+        type: dict
+        keys:
+          server:
+            type: list
```
Comment (jonxstill) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml]:
```suggestion
          server:
            description: SSH host key settings
            type: list
```

**Review by jonxstill — DISMISSED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Review by jonxstill — APPROVED** [review]
LGTM

**Pull request merged**